### PR TITLE
test(pkg): cleanup tests

### PR DESCRIPTION
**Pull request by Alizter**
We replace solve_project in most places with solve, cleanup whitespace and directory creation (not that our helpers do this).

### CODE REVIEW

File: test/blackbox-tests/test-cases/pkg/opam-package-with-extra-source.t [hunk 1]
```diff
@@ -30,21 +30,7 @@ Make a package with an extra-source field and multiple checksums
   > }
   > EOF
 
-  $ opam_repo=$mock_packages
```
Comment (rgrinberg) [inline test/blackbox-tests/test-cases/pkg/opam-package-with-extra-source.t:33]: how come this was deleted?

File: test/blackbox-tests/test-cases/pkg/opam-package-with-subst.t [hunk 2]
```diff
@@ -1,5 +1,4 @@
 We test how opam files with substs fields are translated into the dune.lock file.
-
```
Comment (rgrinberg) [inline test/blackbox-tests/test-cases/pkg/opam-package-with-subst.t:2]: Space here was intentional

File: test/blackbox-tests/test-cases/pkg/opam-package-with-subst.t [hunk 3]
```diff
@@ -41,6 +32,5 @@ The lockfile should contain the substitute action.
   > EOF
 
 The file foo.ml should have been built:
-
```
Comment (rgrinberg) [inline test/blackbox-tests/test-cases/pkg/opam-package-with-subst.t:44]: Intentional space. Could you review all the changes and remove all the unnecessary whitespace tweaks?

File: test/blackbox-tests/test-cases/pkg/opam-package-with-extra-source.t (hunk 1 above)
Comment (Alizter) [reply test/blackbox-tests/test-cases/pkg/opam-package-with-extra-source.t:33]: Because all of this gets done when we do `mkpkg`.

**Comment by Alizter** [comment]
@rgrinberg I've removed all the whitespace changes. Had I known they were all intentional, I wouldn't of touched them.

**Comment by rgrinberg** [comment]
That's why it's important to keep PR's focused on one thing.
